internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -605,6 +621,62 @@ func (r *AgentServiceConfigReconciler) newAgentRoute(ctx context.Context, log lo
 	return route, mutateFn, nil
 }
 
+func (r *AgentServiceConfigReconciler) newAgentIPXERoute(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) (client.Object, controllerutil.MutateFn, error) {
+
+	// Wait for https route to be created first
+	httpsRoute := &routev1.Route{}
+	if err := r.Get(ctx, types.NamespacedName{Name: serviceName, Namespace: r.Namespace}, httpsRoute); err != nil {
+		log.WithError(err).Error("Failed to get https route for agent service")
+		return nil, nil, err
+	}
+	if httpsRoute.Spec.Host == "" {
+		log.Info("Agent https route found, but host not yet set")
+		return nil, nil, nil
+	}
+	log.Infof("Found agent https route '%s'", httpsRoute.Spec.Host)
+
+	weight := int32(100)
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-ipxe", serviceName),
+			Namespace: r.Namespace,
+		},
+		Spec: routev1.RouteSpec{
+			Host: httpsRoute.Spec.Host,
+		},
+	}
+	routeSpec := routev1.RouteSpec{
+		Path: "/api/assisted-install/v2",
```

Comment:
Two issues with that:
* possibly too broad
* hardcoded, not sure if we define this anywhere so that it could be constructed